inst/include/population.h

```diff
@@ -136,6 +140,17 @@ namespace smc {
 
         return *this;
     }
+    
+    /// This function performs a stable calculation of the log sum of the weights, which is useful for
+    /// normalising weights, calculating the effective sample size and estimating the normalising constant.
+    ///
+    /// \param logw The log weights of interest.
+    inline double stableLogSumWeights(const arma::vec & logw){
```

Comment:
Great, thank you. I've created a new source file to contain that and possible other general use function definitions (it didn't fit naturally under the current source files).